mitiq/observable/observable.py

```diff
@@ -154,3 +179,30 @@ def _expectation_from_density_matrix(
 
     def __str__(self) -> str:
         return " + ".join(map(str, self._paulis))
+
+    def __eq__(self, other: Any) -> bool:
+        return np.allclose(self.matrix(), other.matrix())
+
+
+def _combine_duplicate_pauli_strings(
+    paulis: list[PauliString],
+) -> list[PauliString]:
```

Comment:
```suggestion
def _combine_duplicate_pauli_strings(
    paulis: Iterable[PauliString],
) -> List[PauliString]:
```

Looks like `paulis` is only being iterated over, so we can use the typehint to reflect that. You'll have to import `Iterable` from `typing`. Then like Andrea mentioned, `list` is the newer syntax that we aren't currently using so use `typing.List` instead.

If you take this change, you can also remove the cast to list on line 42.